doc/time-series.rst

```diff
@@ -207,3 +210,90 @@ Dataset and DataArray objects with an arbitrary number of dimensions.
 
 For more examples of using grouped operations on a time dimension, see
 :ref:`toy weather data`.
+
+
+.. _NetCDFTimeIndex:
+     
+Non-standard calendars and dates outside the Timestamp-valid range
+------------------------------------------------------------------
+
+Through the optional ``netcdftime`` library and a custom subclass of
+``pandas.Index``, xarray supports a subset of the indexing functionality enabled
+through the standard ``pandas.DatetimeIndex`` for dates from non-standard
+calendars or dates using a standard calendar, but outside the
+`Timestamp-valid range`_ (approximately between years 1678 and 2262).  This
+behavior has not yet been turned on by default; to take advantage of this
+functionality, you must have the ``enable_netcdftimeindex`` option set to
+``True`` within your context (see :py:func:`~xarray.set_options` for more
+information).
+
+For instance, you can create a DataArray indexed by a time
+coordinate with a no-leap calendar within a context manager setting the
+``enable_netcdftimeindex`` option, and the time index will be cast to a
+``NetCDFTimeIndex``:
+
+.. ipython:: python
+
+   from itertools import product
+   from netcdftime import DatetimeNoLeap
+   
+   dates = [DatetimeNoLeap(year, month, 1) for year, month in
+            product(range(1, 3), range(1, 13))]
+   with xr.set_options(enable_netcdftimeindex=True):
+       da = xr.DataArray(np.arange(24), coords=[dates], dims=['time'],
+                         name='foo')
+                         
+.. note::
+
+   With the ``enable_netcdftimeindex`` option activated, a ``NetCDFTimeIndex``
+   will be used for time indexing if any of the following are true:
+
+   - The dates are from a non-standard calendar
+   - Any dates are outside the Timestamp-valid range
+
+   Otherwise a ``pandas.DatetimeIndex`` will be used.  In addition, if any
+   variable (not just an index variable) is encoded using a non-standard
+   calendar, its times will be decoded into ``netcdftime.datetime`` objects,
+   regardless of whether or not they can be represented using
+   ``np.datetime64[ns]`` objects.
+                         
+For data indexed by a ``NetCDFTimeIndex`` xarray currently supports `partial
```

Comment:
Maybe change this to bullet points rather than one long sentence? I.e. "...currently supports: " followed by one bullet each for partial datetime string indexing, dt accessor, groupby, and serialization.